shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1238,6 +1238,9 @@ bool WebContents::PlatformHandleKeyboardEvent(
 content::KeyboardEventProcessingResult WebContents::PreHandleKeyboardEvent(
     content::WebContents* source,
     const content::NativeWebKeyboardEvent& event) {
+  if (exclusive_access_manager_->HandleUserKeyEvent(event))
+    return content::KeyboardEventProcessingResult::HANDLED;
```

Comment:
Just pondering, perhaps we should allow an app to override this behavior? Not sure what the best thing to do there is. I like that this works as probably-expected "by default", but also it seems potentially nice for an app to be able to choose its own method of exiting pointer lock. e.g. for a game written with electron, it might be good to allow the in-game menu to be shown with [Esc], but if pointer lock is automatically escaped by that, then that would instead drop pointer lock? Is that sort of problem fixable from the web content side with the keyboard lock stuff?

Reply:
I also thought about this a bit - I think that it's probably easier and more intuitive to stick to [what Chromium does](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/exclusive_access/exclusive_access_manager.cc;l=109-128;drc=054e08864177603f17edbc111db7ebc8586906bd;bpv=0;bpt=1) here for esc key behavior, given that the developer can also capture keydown to trigger `document.exitPointerLock()` - to handle this the way you describe i think we'd have to extend or patch some of the exclusive access code since it's tied into multiple controllers and bits of state 🤔

Reply:
@codebytere hm, i think you're describing allowing the developer to exit pointer lock using other keys, but can the developer _prevent_ <kbd>Esc</kbd> from exiting pointer lock?

Reply:
@nornagon no, not with this PR. Given that this aligns with Chromium, though, I think for the purposes of this PR that would be preferable?

Reply:
@nornagon do you think you'd be ok proceeding as-is?